### PR TITLE
Workaround Issue 17081 - Bodies in extern cpp functions in D files are not linked

### PR DESCRIPTION
**Pull request by Darredevil**
https://issues.dlang.org/show_bug.cgi?id=17081

### CODE REVIEW

**Comment by kinke** [comment]
Why do you implement these functions? Isn't adding the `const` qualifier to `name()` enough to fix the linking error? Additionally, you don't check whether `_arg` for `before()` is null, but claim that the function is `nothrow`...

**Comment by Darredevil** [comment]
@kinke adding `const` was not enough to fix the problem, see the bug tracker link for example. Implementing them was the only way to fix the problem. 
As for not checking for null, it's because I translated the exact source code from the GNU stdcpp, however you are correct, the `nothrow` will be removed.

File: src/core/stdcpp/typeinfo.d [hunk 1]
```diff
@@ -117,8 +117,15 @@ else version (CRuntime_Glibc)
         {
             void dtor1();                           // consume destructor slot in vtbl[]
             void dtor2();                           // consume destructor slot in vtbl[]
-            final const(char)* name();
-            final bool before(const type_info) const;
+            final const(char)* name()() const nothrow {
+                return _name[0] == '*' ? _name + 1 : _name;
+            }
+            final bool before()(const type_info _arg) const {
+                import core.stdc.string:strcmp;
```
Comment (andralex) [inline src/core/stdcpp/typeinfo.d]: spaces around `:`, cc @wilzbach

File: src/core/stdcpp/typeinfo.d [hunk 2]
```diff
@@ -117,8 +117,15 @@ else version (CRuntime_Glibc)
...
+                return (_name[0] == '*' && _arg._name[0] == '*')
+                    ? _name < _arg._name
+                    : strcmp (_name, _arg._name) < 0;
```
Comment (andralex) [inline src/core/stdcpp/typeinfo.d]: no space before `(` in function call (i.e. use math-like notation), cc @wilzbach

**Comment by andralex** [comment]
BTW this shall not be considered a fix, so I'll change the title.

**Comment by Darredevil** [comment]
@andralex Fixed spacing, waiting final tests for merge

**Comment by andralex** [comment]
Auto-merge toggled on

**Comment by andralex** [comment]
@Darredevil great, thx. Next step would be probably to actually see the problem with https://issues.dlang.org/show_bug.cgi?id=17081. 

Also please push this again, looks like the autotester failed for no cause. cc @MartinNowak @braddr

**Comment by kinke** [comment]
@andralex: Related & trivial, but sitting there for more than a month already and mentioning more C++ interop issues: #1709

**Pull request merged**